code/datums/action.dm

```diff
@@ -43,10 +43,16 @@
 				return
 			Remove(owner)
 		owner = M
+		var/button_id = 1
+		for(var/datum/action/A in M.actions)
+			if(A.name == name)
+				button_id += 1 //counting actions with same name to assign special name to button ([name]_[id]) for positioning memory
+		button.id = "[name]_[button_id]"
```

Comment:
this will fail, really quickly.

Reply:
For some reasons didn't noticed it before you pointed it out.